src/T2DMap.cpp

```diff
@@ -1923,6 +1958,63 @@ void T2DMap::paintAreaExits(QPainter& painter, QPen& pen, QList<int>& exitList,
                     }
                     painter.drawPolyline(polyLinePoints.data(), polyLinePoints.size());
 
+                    if (!getUserDataBool(room->userData, ROOM_UI_DONTSHOWAREAEXITNAME, false)) {
+                        QPointF areaExitNameOffset = parseOffsetString(room->userData.value(ROOM_UI_AREAEXITNAMEOFFSET + "." + itk.key()));
+                        int exitRoomId = -1;
+                        QMapIterator<int, QString> otherExitIt = room->getOtherMap();
+                        while (otherExitIt.hasNext()) {
+                            otherExitIt.next();
+                            if (otherExitIt.value().startsWith(QLatin1String("0")) || otherExitIt.value().startsWith(QLatin1String("1"))) {
+                                if (otherExitIt.value().mid(1) == itk.key()) {
+                                    exitRoomId = otherExitIt.key();
+                                    break;
+                                }
+                            } else if (otherExitIt.value() == itk.key()) {
+                                exitRoomId = otherExitIt.key();
+                                break;
+                            }
+                        }
+                        if (exitRoomId > -1) {
```

Comment:
For forward compatibility it will be safer to assume that a valid exit ID is `> 0` rather than `> -1` - `0` is not currently valid anyhow - and I intend to ensure that is the case in the future...! :grin:

Reply:
OK, fix it